### PR TITLE
fix(nemesis): MultipleHardRebootNode: don't verify the rebooted node through itself (SCT-953)

### PR DESCRIPTION
**Pull request by aleksbykov**
`MultipleHardRebootNodeMonkey` verifies the node it has just hard-rebooted by calling
`wait_for_nodes_up_and_normal(nodes=[self.target_node])` without a verification node.
`check_nodes_up_and_normal` forwards `verification_node=None` down to `get_nodetool_status`,
which falls back to `random.choice(self.nodes)` over *every* node of the cluster - including
the target node that is still booting. On the reported run the roll landed on the target
itself, all three inner retries re-rolled onto it again, and the wait was abandoned ~37
seconds before the node actually reached NORMAL.

This PR fixes only that nemesis: a disruptive nemesis must never ask its own target node
about the state of the cluster. A peer is reserved from the nemesis node allocator around
each up-and-normal check and the verification goes through it. The target node stays in
`active_nemesis_on_nodes` for the whole disruption, so `select_target_node` can never hand
it back. The reservation is taken and released per reboot iteration rather than held for the
whole loop, matching how `node_isolation` obtains a working node.

The unsafe `random.choice(self.nodes)` default in `get_nodetool_status` is a framework-wide
problem that affects every caller that omits `verification_node`, not only this nemesis. It
is addressed separately in a follow-up PR; this one stays small so the sanity jobs that
`MultipleHardRebootNodeMonkey` is currently breaking can be unblocked with a limited
testing surface.

Add unit tests for this nemesis, which had none. One drives a real `NemesisNodeAllocator` to
cover the guarantee that the peer is never the target; the other two assert on how the
allocator is called.

### Testing

- [ job passed](https://argus.scylladb.com/tests/scylla-cluster-tests/c3ad76b7-63f4-4559-b787-de8a3bd9f7c0)

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: sdcm/nemesis/__init__.py [hunk 1]
```diff
@@ -834,7 +834,13 @@ def disrupt_multiple_hard_reboot_node(self) -> None:
             else:
                 self.log.info("Waiting JMX services to start after node reboot")
                 self.target_node.wait_jmx_up()
-            self.cluster.wait_for_nodes_up_and_normal(nodes=[self.target_node])
```
Comment (pehala) [inline sdcm/nemesis/__init__.py:837]: Isnt there a bug in this function instead? Based on the name this seems like exactly what we want.

Comment (cezarmoise) [reply sdcm/nemesis/__init__.py:837]: Inside `get_nodetool_status` there is a block
```python
if not verification_node:
    verification_node = random.choice(self.nodes)
```
Maybe the change could be there, similar to https://github.com/scylladb/scylla-cluster-tests/pull/11916

Comment (aleksbykov) [reply sdcm/nemesis/__init__.py:837]: I think about it, but decided to create fast small fix for nemesis which breaking the sanity jobs. Also in this case it won't require to many testing.
fixing get_nodetool_status could affect too many places, and may be better to have another large pr with research and investigation. These methods are used in many places with different parameters.

Comment (pehala) [reply sdcm/nemesis/__init__.py:837]: But the problem is that the problem you describe ("Scylla serves HTTP 404 on /storage_service/load_map until storage_service registers its REST routes, so nodetool exited 4") is general and having workaround in single nemesis is not sufficient. IF what you write is true, then it is a widespread problem that can happen anywhere and should happen much more often.

Comment (aleksbykov) [reply sdcm/nemesis/__init__.py:837]: ok, i will prepare new Fix which should fix root issue. This also could be merged, because it is good practice to choose verification node inside nemesis. especially in disruptive nemesis

**Review by fruch — APPROVED** [review]
